pkg/apis/k0s/v1beta1/api.go

```diff
@@ -72,6 +73,17 @@ func DefaultAPISpec() *APISpec {
 	return a
 }
 
+func (a *APISpec) LocalURL() *url.URL {
+	var host string
+	if a.OnlyBindToAddress {
+		host = net.JoinHostPort(a.Address, strconv.Itoa(a.Port))
+	} else {
+		host = fmt.Sprintf("localhost:%d", a.Port)
```

Comment:
Why not use `net.JoinHostPort` in this case too?

Reply:
We could, but we don't need to. JoinHostPort is only necessary if the host may be an IPv6 address, which, in this case, it isn't. Would you prefer to use it anyways?